src/filter_iterator.rs

```diff
@@ -5,20 +5,34 @@ use crate::SlateDBError;
 
 pub(crate) struct FilterIterator<T: KeyValueIterator> {
     iterator: T,
-    predicate: Box<dyn Fn(&RowEntry) -> bool + Send>,
+    predicate: Box<dyn Fn(&RowEntry) -> bool + Send + Sync>,
 }
 
 impl<T: KeyValueIterator> FilterIterator<T> {
-    pub(crate) fn new(iterator: T, predicate: Box<dyn Fn(&RowEntry) -> bool + Send>) -> Self {
+    pub(crate) fn new(
+        iterator: T,
+        predicate: Box<dyn Fn(&RowEntry) -> bool + Send + Sync>,
+    ) -> Self {
         Self {
             predicate,
             iterator,
         }
     }
 
-    pub(crate) fn wrap_ttl_filter_iterator(iterator: T, now: i64) -> Self {
-        let filter_entry = move |entry: &RowEntry| is_not_expired(entry, now);
-        Self::new(iterator, Box::new(filter_entry))
+    #[allow(unused)]
+    pub(crate) fn new_with_ttl_now(iterator: T, ttl_now: i64) -> Self {
```

Comment:
if you want to just remove this for now since it's unused that's ok with me, since I can just add it back if/when I need it for the #scan ttl expiration PR. But it shouldn't be too long until then so it also seems fine leaving it here for now, your call 🤷‍♀️

Reply:
i feel this method will be useful in the next change, so i think this `#[allow[unused]]` will not last long 😲